src/engine/tt_hash_table.h

```diff
@@ -46,40 +47,51 @@ class TtHashTable {
     using ProbeResult = std::pair<int32_t, movegen::Move>;
     constexpr static std::optional<ProbeResult> probe(uint64_t key, int32_t alpha, int32_t beta, uint8_t depth, uint8_t ply)
     {
-        auto& entry = s_ttHashTable[key % s_ttHashSize];
+        auto& entryWrapper = s_ttHashTable[key % s_ttHashSize];
+        // Load all at once atomically
+        auto oldEntry = entryWrapper.load(std::memory_order_acquire);
```

Comment:
Rename to `loadedEntry`. Pattern makes sense though: load once so not updating during control flow.